board.cpp

```diff
@@ -0,0 +1,205 @@
+#include "board.h"
+const int undefinedPerspective = -1;
+
+bool Position::operator==(const Position &b) const{
+    return ((xpos == b.xpos)
+        && (ypos == b.ypos));
+}
+
+bool Mine::operator==(const Mine &b) const{
+    return ((xpos == b.xpos)
+        && (ypos == b.ypos))
+        && (owner == b.owner);
+}
+
+bool Mine::operator==(const Position &b) const{
+    return ((xpos == b.xpos)
+        && (ypos == b.ypos));
+}
+
+Board::~Board(){
+    delete(players);
+}
+
+bool isPositionValid(Board const& board, Position pos){ //check if pos if withing acceptable values 
+    //If the position is outside the board, return false;
+    if((pos.xpos < 1)
+    || (pos.ypos < 1)
+    || (pos.xpos > board.width)
+    || (pos.ypos > board.height)) return false;
+    //if the position is disabled, return false
+    for(int i = 0; i < board.disabledPositions.size(); i++){
+        if(pos == board.disabledPositions[i]) return false;
+    }
+    //else, return true
+    return true;
+}
+
+//gets all possible valid positions
+//mainly for bot logic
+std::vector<Position> getValidTiles(Board const& board){ 
+    std::vector<Position> validPositions;
+    for(int x = 1; x <= board.width; x++){
+        for(int y = 1; y <= board.height; y++){
+            Position pos;
+            pos.xpos = x;
+            pos.ypos = y;
+            if (isPositionValid(board, pos)){
+                validPositions.push_back(pos);
+            }
+        }
+    }
+    return validPositions;
+}
+
+int getValuesWithinRange(std::string prompt, int min, int max){ //function to input an int value within range, used for board initialization, probably could be reporopused for other uses
+    int ret = min-1;
+    while ((ret < min) || (ret > max)){
+        std::cout << prompt << std::endl;
+        std::cout << "value must be between " << min << " and " << max << std::endl;
+        std::cin >> ret;
+    }
+    return ret;
+    
+}
+//board construction
+Board createBoard(){
+    Board board;
+    bool created = false;
+    int width = NULL;
+    int height = NULL;

```

Comment:
#must not use `NULL` to initialize non-pointer types.

Actually... #must not use `NULL` at all; use `nullptr` instead (which would be a compile error in this case - because you're not meant to do that!).